src/silx/io/h5py_utils.py

```diff
@@ -363,12 +390,7 @@ def __init__(
         """
         # File locking behavior has changed in recent versions of libhdf5
         if HDF5_HAS_LOCKING_ARGUMENT != H5PY_HAS_LOCKING_ARGUMENT:
-            _logger.critical(
-                "The version of libhdf5 ({}) used by h5py ({}) is not supported: "
-                "Do not expect file locking to work.".format(
-                    h5py.version.hdf5_version, h5py.version.version
-                )
-            )
+            warm_incoherent_locking_argument()
```

Comment:
I agree this makes it more complicated to have a function to log a specific message